apps/assistants/views.py

```diff
@@ -140,6 +149,31 @@ def form_valid(self, form):
         return response
 
 
+def check_sync_status(request, team_slug, pk):
+    assistant = get_object_or_404(OpenAiAssistant, team=request.team, pk=pk)
+    try:
+        is_synced = is_synced_with_openai(assistant)
+    except OpenAiSyncError:
+        is_synced = False
+    context = {
+        "is_synced": is_synced,
+        "object": assistant,
+    }
+    return render(request, "assistants/sync_status.html", context)
+
+
+class SyncEditingOpenAiAssistant(BaseOpenAiAssistantView, View):
+    permission_required = "assistants.change_openaiassistant"
+
+    def post(self, request, team_slug: str, pk: int):
+        assistant = get_object_or_404(OpenAiAssistant, team=request.team, pk=pk)
+        try:
+            sync_from_openai(assistant)
+        except OpenAiSyncError as e:
+            messages.error(request, f"Error syncing assistant: {e}")
+        return check_sync_status(request, team_slug, pk)
```

Comment:
Instead of re-doing the check here, this should probably do a page reload so that any changes that were pulled in are also reflected in the form. This will also then re-trigger the sync check on page load.

You can do that by setting the `HX-Refresh` header to true:

```suggestion
        return HttpResponse(headers={"HX-Refresh": "true"})
```